File: Node/Node.py
class Node:
    def __init__(self, data):
        self.data = data 
        self.next = None
    def setNext(self,next):
        self.next = next
    def getNext(self):
        return self.next
    
if __name__ == "__main__":
    node1 = Node(1)
    node2 = Node(2)
    node3 = Node(3)
    node1.setNext(node2)
    node2.setNext(node3)
    node_tmp = node1 
    while node_tmp:
        print(node_tmp.data)
        node_tmp = node_tmp.next 

File: Node/Node.cpp
#include "Node.h"
#include <iostream>
using namespace std;

Node::Node(){
    this->data = -1;
    this->next = nullptr;
}

Node::Node(int data){
    this->data = data;
    this->next = nullptr;
}

int Node::getData(){
    return this->data;
}

Node *Node::getNext(){
    return this->next;
}
void Node::setData(int data){
    this->data = data;
}
void Node::setNext(Node *next){
    this->next = next;
}

int main(){
    Node node1(1);
    Node node2(2);
    Node node3(3);
    Node node4(4);
    node1.setNext(&node2);
    node2.setNext(&node3);
    node3.setNext(&node4);
    Node *tmp = &node1;
    while (tmp){
        cout << tmp->getData() << endl;
        tmp = tmp->getNext();
    }
    return 0;
}

